neotron-bmc-pico/src/main.rs

```diff
@@ -583,7 +588,37 @@ mod app {
 					);
 				}
 			}
-			// TODO: Read ADC for 3.3V and 5.0V rails and check good
+			// TODO: Also monitor 5.0V rail
+
+			// Wait for power good
+			if ctx.shared.state_dc_power_enabled.lock(|r| *r) == DcPowerState::Starting {
+				let mon_3v3 = ctx.local.adc.read_abs_mv(ctx.local.pin_3v3_monitor);
+				defmt::trace!("3v3 reading: {}", mon_3v3);
+				if mon_3v3 < 1600 {
+					defmt::info!("3v3 below threshold: {} mv", mon_3v3);
+				} else {
+					defmt::info!(
+						"Power good. 3v3 at {} mV. Continue with startup sequence.",
+						mon_3v3
+					);
+					// Change the power state machine to "On". We were in 'Starting' to ignore
+					// any further button events until the button had been
+					// released and to wait for power good.
+					ctx.shared
+						.state_dc_power_enabled
+						.lock(|r| *r = DcPowerState::On);
+					// Steps 1 - 4 happened above, in the button press handler
+					// Step 5 - Leave it in reset for a while.
+					// TODO: Start monitoring 3.3V and 5.0V rails here
```

Comment:
This comment was mindlessly copied and should be removed. I'll update the PR later.